core/string/translation_server.cpp

```diff
@@ -295,14 +303,46 @@ String TranslationServer::get_locale_name(const String &p_locale) const {
 		}
 	}
 
-	String name = get_language_name(lang_name);
-	if (!script_name.is_empty()) {
-		name = name + " (" + get_script_name(script_name) + ")";
-	}
-	if (!country_name.is_empty()) {
-		name = name + ", " + get_country_name(country_name);
+	if (p_native) {
+		String name;
+		if (!country_name.is_empty()) {
+			if (native_locale_map.has(lang_name + "_" + country_name)) {
```

Comment:
You could use `find()`/`getptr()` to avoid double lookup (and constructing the key twice...)